tests/utils/PolygonTest.cpp

```diff
@@ -305,18 +326,263 @@ TEST_F(PolygonTest, simplifyLimitedError)
+        long_segment *= -1;
+        if (height < max_height)
+        {
+            simplifiable_bend_count++;
+        }
+    }
+    zigzag.emplace_back(zigzag.back() + Point(-non_simplifiable_bound, -non_simplifiable_bound)); // complete polygon with a point which enforces non-simplification to both its segments
+
+    Polygon zigzag_before = zigzag;
+
+    zigzag_polygons.simplify(non_simplifiable_bound, max_height);
+
+    if (visualize)
+    {
+        SVG svg("output/simplifyIncreasingLimitedError.svg", AABB(zigzag_before));
+        svg.writePolygon(zigzag_before);
+        svg.nextLayer();
+        svg.writePolygon(zigzag, SVG::Color::RED);
+    }
+    EXPECT_THAT(zigzag.size(), testing::Eq(zigzag_before.size() - simplifiable_bend_count)) << "Should simplify bends with height 100 up to 500";
+}
+
+
+TEST_F(PolygonTest, simplifySineLimitedError)
+{
+    // Generate a straight line with sinusoidal errors which should be simplified back into a more straight line
+    
+    // Hypothetically simplify() might replace each half period of the sine with a straight segment,
+    // but because simplify() is heuristic it introduces more segments.
+    // The function signature doesn't provide any guarantee about how much simplification will occur,
+    // but in practice it should at least simplify up to double the minimal segment count.
+    
+    Polygons sine_polygons;
+    PolygonRef sine = sine_polygons.newPoly();
+    
+    constexpr coord_t length = 10000;
+    constexpr coord_t deviation = 500;
+    constexpr size_t bulge_count = 5;
+    
+    sine.emplace_back(length, 0);
+    sine.emplace_back(length, length);
+    sine.emplace_back(0, length);
+    sine.emplace_back(0, 0);
+
+    for (coord_t x = 100; x < length; x += 100)
+    {
+        sine.emplace_back(x, std::sin(INT2MM(x) / INT2MM(length) * M_PI * bulge_count ) * deviation);
+    }
+    Polygon sine_before = sine;
 
-    //We want it to not merge the lines 1400 and 1500 any more, but do merge all lines before it.
-    //Take the area of the 1400 by 1500 and plug it into the formula for the height to get at the baseline height, which is our allowed error.
-    constexpr coord_t area = 1400 * 1500 / 2;
-    const coord_t diagonal_length = std::sqrt(1400 * 1400 + 1500 * 1500); //Pythagoras.
-    //A = 0.5 * b * h. diagonal_length is the base line in this case.
-    //2A = b * h
-    //2A / b = h
-    const coord_t height = 4 * area / diagonal_length; //Error of the first vertex we want to keep, so we must set the limit to something slightly lower than this.
-    spiral_polygons.simplify(999999999, height - 10);
-
-    EXPECT_THAT(spiral.size(), testing::AllOf(testing::Ge(11 - 5), testing::Le(11 - 4))) << "Should merge segments of length 1000 through 1400 and (optionally) first with last.";
+    sine_polygons.simplify(length / 2, 2 * deviation);
+    if (visualize)
+    {
+        SVG svg("output/simplifySineLimitedError.svg", AABB(sine_before));
```

Comment:
There is a lot of code duplication with writing these svg files. I think we should convert it into a re-usable function

Reply:
Well. It's not like there's any complexity in the code. Nothing can go wrong. It would maybe be a bit cleaner.. Sure,

Reply:
Yeah, it's a bit nitpicking, but I like my code to be clean :)

Reply:
The conceptual load of reading the current code is no more, or perhaps even less than reading a new function which fits on one line. I don't think the physical length of the code should be a criterion to decide on code-cleanliness.

I think it is nitpicking. Let's not discuss. I would be okay with you fixing this, but I am overburdened atm.